ci/gitlab/api.py

```diff
@@ -154,8 +154,8 @@ def get_repos(self, session):
         return owner_repo
 
     @copydoc(GitAPI.get_branches)
-    def get_branches(self, owner, repo):
-        url = "%s/repository/branches" % (self._repo_url(owner, repo))
+    def get_branches(self, path_with_namespace):
```

Comment:
So I think this might be a problem. There is a general API that is used across all 3 git servers (GitHub, Bitbucket, and GitLab). Having this one a different signature could cause a problem.
But it looks like we don't use "get_branches" anywhere? Could be a relic of when we used the API differently.
Maybe this whole function can be dropped (on all 3)?

Reply:
We do? For all three... hmmm. Well, unittests pass. So I assume all three succeeded. But I'll look into it a bit more.

Reply:
Yeah, see `ci/git_api.py`. It allows the rest of Civet to not care about what server we are talking to.
The tests pass because apparently we don't have a test that goes through the `get_branches` abstract API, probably because it isn't used anywhere.